lang/java/avro/src/main/java/org/apache/avro/io/JsonDecoder.java

```diff
@@ -471,7 +464,11 @@ public Symbol doAction(Symbol input, Symbol top) throws IOException {
             if (currentReorderBuffer == null) {
               currentReorderBuffer = new ReorderBuffer();
             }
-            currentReorderBuffer.savedFields.put(fn, getValueAsTree(in));
+            TokenBuffer tokenBuffer = new TokenBuffer(in);
```

Comment:
Can we wrap a try-with-resources around the `tokenBuffer`, since it is `AutoClosable`.